src/api/java/com/minecolonies/api/tileentities/TileEntityRack.java

```diff
@@ -438,11 +428,7 @@ else if (getOtherChest() != null)
             {
                 if (isMain())
                 {
-                    if (combinedHandler == null)
-                    {
-                        combinedHandler = new CombinedInvWrapper(inventory, getOtherChest().getInventory());
-                    }
-                    return LazyOptional.of(() -> (T) combinedHandler);
+                    return LazyOptional.of(() -> (T) new CombinedItemHandler(RACK, inventory, getOtherChest().getInventory()));
```

Comment:
Nope.
The idea of lazy optionals is that they should only be created when needed.

Meaning that these optionals should be effectively final as long as they are valid (aka as long as the block persists and his "neighbor"-state did not change.
Once any of these above properties change, you invalidate the old lazy optional and save a new one in a field which you then return in the getCapabilities method.

THis ensures that mods like AE etc can hold on to the itemhandler for as long as he is valid and only query for a new one once we invalidate the old one.